### PR TITLE
Added UI YAML Support

### PR DESCRIPTION
**Pull request by dropmann**

### CODE REVIEW

File: ui.template [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
+# This file is an automatically generated template. It will NOT be subsequently
+# overwritten by the compiler and may be edited
```
Comment (jonathon-love) [inline ui.template]: so the file generated will be overwritten, but not replaced. you can probably just remove these two lines altogether.

File: uicompiler.js [hunk 2]
```diff
@@ -39,43 +65,167 @@ const createUIElements = function(options) {
             var neededGroup = groupConstructors.getAppropriateSupplier(item);
 
             if (neededGroup !== currentGroup || areControlsGroupCompatible(lastCtrlType, item.type) === false) {
-                indentCount = 2;
+                indentCount = 1;
```
Comment (jonathon-love) [inline uicompiler.js]: so it appears you're constructing the yaml manually, but wouldn't it be better to simply construct the objects, and then do a `yaml.safeDump()`

creating your own yaml won't really work with round-tripping either. i.e. reading in a YAML, inserting a new item into it, and writing it back out again.

File: ui.template (hunk 1 above)
Comment (jonathon-love) [inline ui.template]: bump

**Comment by jonathon-love** [comment]
wait, i was under the impression you'd made a bunch of changes to the compiler ... but this is still the same as yesterday.

**Comment by dropmann** [comment]
no this is the latest stuff. What are you comparing too?

**Comment by jonathon-love** [comment]
oh OK, but you're still using the ui template, which shouldn't be needed any more?

**Comment by dropmann** [comment]
No i do need it to maintain the readablity of the file. Otherwise it makes in alphabetical order. I am make some other changes so hold of doing anything for a bit :)

**Comment by jonathon-love** [comment]
are you sure about the alphabetical order thing? it doesn't do that to me.

**Comment by jonathon-love** [comment]
(as in, when i use `yaml.safeDump()` it preserves the order things were added)

**Comment by jonathon-love** [comment]
(which i'm well pleased with)

**Comment by dropmann** [comment]
You were right. I have committed everything again.

File: uicompiler.js [hunk 3]
```diff
@@ -112,173 +409,182 @@ const groupConstructors = {
...
-        ctrl += indent + '   showColumnHeaders: false,\n'
-        ctrl += indent + '   fullRowSelect: true,\n'
-        ctrl += indent + '   columns: [\n'
-        ctrl += indent + '       { type: "listitem.variablelabel", name: "column1", label: "", stretchFactor: 1 }\n';
-        ctrl += indent + '   ]\n'
-        ctrl += indent + "}";
+    Variables: function(item) {
+        var ctrl = { };
+
+        ctrl.type = "variabletargetlistbox";
```
Comment (jonathon-love) [inline uicompiler.js]: did you have any thoughts on making this camel case? normally i'd let this slide, but this is going into our public API.

Comment (jonathon-love) [inline uicompiler.js]: by this, i mean *these*

**Pull request merged**